### PR TITLE
#KT-9594 J2K: null cast incorrect conversion. Fix

### PR DESCRIPTION
**Pull request by AlexTrask**
Any criticism is welcome

### CODE REVIEW

**Comment by semoro** [comment]
This branch contains commits from another PR, please fix it

File: j2k/src/org/jetbrains/kotlin/j2k/ExpressionConverter.kt [hunk 1]
```diff
@@ -650,7 +650,7 @@ class DefaultExpressionConverter : JavaElementVisitor(), ExpressionConverter {
             result = MethodCallExpression.buildNonNull(operandConverted, typeConversion)
         }
         else {
-            val nullability = if (operandConverted.isNullable && !expression.isQualifier())
+            val nullability = if ((operandConverted.isNullable && !expression.isQualifier()) || operandType == PsiType.NULL)
```
Comment (semoro) [inline j2k/src/org/jetbrains/kotlin/j2k/ExpressionConverter.kt]: Better not to hack this check, but fix problem with `operandConverted.isNullable == false` for `null` literal

Comment (AlexTrask) [inline j2k/src/org/jetbrains/kotlin/j2k/ExpressionConverter.kt]: okey, understood your idea. I will apply your comment

**Comment by AlexTrask** [comment]
Apply your comments, but commit it as 'push -f'

**Comment by semoro** [comment]
Merged manually. 5edb872224f9d31d7d379b7504f9e5a9d508ff12
Thanks for your contribution! :blush:

**Pull request closed without merging**